Ardupilog.m

```diff
@@ -427,41 +419,47 @@ function delete(obj)
             slice.numMsgs = 0;
             
             % Loop through the LogMsgGroups, slicing each one
-            for msgGroup = slice.logMsgGroups
-                % If the LogMsgGroup has been deleted, skip it
-                if ~isvalid(msgGroup)
-                    continue
+            logProps = properties(obj);
+            for i = 1:length(logProps)
+                propertyName = logProps{i}; % Get the name of the property under examination
+                % We are interested only in LogMsgGroup objects, skip the rest of the properties
+                if ~isa(obj.(propertyName),'LogMsgGroup') 
+                    continue;
                 end
                 
                 % Slice the LogMsgGroup
-                lmg_slice = slice.(msgGroup.Name).getSlice(slice_values, slice_type);
+                lmg_slice = slice.(propertyName).getSlice(slice_values, slice_type);
                 % If the slice is not empty, add it to the Ardupilog slice
                 if isempty(lmg_slice)
-                    delete(msgGroup)
+                    delete(slice.(propertyName))
```

Comment:
I just glanced at #54 ... was that PR addressing this idea? If so, we can move the discussion there... I'm a bit slow on my code reviewing this weekend